runelite-client/src/main/java/net/runelite/client/plugins/banktags/tabs/TabInterface.java

```diff
@@ -720,6 +726,56 @@ private void deleteTab(String tag)
 		scrollTab(0);
 	}
 
+	private void renameTab(String oldTag)
+	{
+		chatboxPanelManager.openTextInput("Enter new tag name for tag \"" + oldTag + "\":")
+				.onDone((newTag) -> clientThread.invoke(() ->
+				{
+					if (!Strings.isNullOrEmpty(newTag) && !newTag.equalsIgnoreCase(oldTag))
+					{
+						if (tabManager.find(newTag) == null)
+						{
+							loadTab(newTag);
```

Comment:
`line 738` to `line 754` can all be replaced if you 

1. Just rename the tab
2. Set the icon in config
3. Delete the old icon
4. Open new tag if old tag is selected